crates/bevy_reflect/src/fields.rs

```diff
@@ -0,0 +1,54 @@
+use crate::{Reflect, TypeIdentity};
+use std::borrow::{Borrow, Cow};
+
+/// The named field of a reflected struct
+#[derive(Clone, Debug)]
+pub struct NamedField {
+    name: Cow<'static, str>,
+    id: TypeIdentity,
+}
+
+impl NamedField {
+    pub fn new<T: Reflect>(name: &str) -> Self {
```

Comment:
Can we have an alternative constructor that takes a &'static str, and uses Cow::Borrowed instead? Avoids an allocation here.